src/Castle.Facilities.Synchronize/CreateOnUIThreadInspector.cs

```diff
@@ -171,8 +173,11 @@ private static IReference<IProxyGenerationHook> ObtainProxyHook(IConfiguration c
 						var message = String.Format("The specified controlProxyHook does " +
 						                            "not implement the interface {1}. Type {0}",
 						                            hookType.FullName, typeof(IProxyGenerationHook).FullName);
-
+#if FEATURE_SYSTEM_CONFIGURATION
 						throw new ConfigurationErrorsException(message);
+#else
+						throw new InvalidOperationException(message);
+#endif
```

Comment:
@jonorossi What would be a good exception to throw here when `ConfigurationErrorsException` is not available?